pandas/tests/scalar/period/test_period.py

```diff
@@ -632,6 +632,13 @@ def _ex(p):
         result = p.to_timestamp("5S", how="start")
         assert result == expected
 
+    def test_to_timestamp_business_end(self):
+        per = pd.Period("1990-01-05", "B")  # Friday
```

Comment:
can you test how='E', and is this result correct?

Reply:
we have tests for the other case, and it is the correct result.

i got this test by going into the PeriodArray.to_timestamp code and computing the same thing element-wise, then asserting that they matched.  this is one of the cases that failed

Reply:
I don't think this is correct, 1/6 is a sunday, by-definition not a business day

Reply:
you are missing my point, should this not be ``pd.Timestamp('1990-01-8') - pd.Timedelta(nanoseconds=1)``, IOW 1 nano less than the next period which *starts* on 1990-01-8

e.g. (not including this PR)
```
In [1]: p = pd.Period("1990-01-05", "B")                                                                                                                      

In [2]: p                                                                                                                                                     
Out[2]: Period('1990-01-05', 'B')

In [3]: p.start_time                                                                                                                                          
Out[3]: Timestamp('1990-01-05 00:00:00')

In [4]: p.end_time                                                                                                                                            
Out[4]: Timestamp('1990-01-07 23:59:59.999999999')

In [6]: pd.Timestamp('1990-01-6') - pd.Timedelta(nanoseconds=1)                                                                                               
Out[6]: Timestamp('1990-01-05 23:59:59.999999999')

In [8]: pd.Timestamp('1990-01-8') - pd.Timedelta(nanoseconds=1)                                                                                               
Out[8]: Timestamp('1990-01-07 23:59:59.999999999')
```

I believe we should be making [4] == [8]

[6] (what you are testing) does not seem right as it *does*n't include the weekend. I believe we cover the complete space with the end-times, IOW you can line the periods up and there is NO space in-between.